### PR TITLE
DSEGOG-68 add husky & lint-staged to lint files on commit

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
I've had a few commits fail the linting CI check, and this can be avoided by running the linters on commit via lint-staged & husky. I've tested that it lets you commit a valid file without complaining as well as not letting you commit an invalid file.

Also, I've copied the eslintrc file from SG/DG with some minor changes, this applies more linting rules such as cypress/typescript rules.

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] Try committing files that break the linting rules (easiest is to add an unused import/variable)

## Agile board tracking
Closes DSEGOG-68

### CODE REVIEW

File: src/plotting/plotList.component.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ const PlotCard = (props: { plotTitle: string; plotConfig: PlotConfig }) => {
   );
 };
 
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
 interface PlotListProps {}
```
Comment (sam-glendenning) [inline src/plotting/plotList.component.tsx:42]: Why do we have this here again?

Comment (louise-davies) [reply src/plotting/plotList.component.tsx:42]: It's just a standard thing - makes it easier to add new props this way. Although I'm not sure whether we'd ever have to pass props to this, so perhaps I should just delete it

Comment (sam-glendenning) [reply src/plotting/plotList.component.tsx:42]: You can if you want, it's not a major change or anything, otherwise I'm happy to approve it

**Review by sam-glendenning — APPROVED** [review]

File: src/plotting/plotList.component.tsx (hunk 1 above)
Comment (louise-davies) [reply src/plotting/plotList.component.tsx:42]: I'll leave it then - we can always remove later if we really know we'll never need to pass props

**Pull request merged**